request.go

```diff
@@ -144,3 +144,58 @@ func (r *ClientRequest) DoJSON(ctx context.Context, model any) error {
 
 	return json.NewDecoder(resp.Body).Decode(model)
 }
+
+// DoRaw exposes the underlying io.ReadCloser for the response body.
+// The caller is responsible for closing the ReadCloser and unmarshaling the
+// results.
+func (r *ClientRequest) DoRaw(ctx context.Context) (io.ReadCloser, error) {
```

Comment:
This method allows the user to stream from the response body directly, enabling users to avoid situations where large archivist objects are entirely loaded into memory.